tests/bugs/replicate/issue-3238-mdata-unnecessary-heal.t

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+. $(dirname $0)/../../include.rc
+. $(dirname $0)/../../volume.rc
+
+cleanup;
+
+TEST glusterd
+TEST pidof glusterd
+TEST $CLI volume create $V0 replica 2 arbiter 1 $H0:$B0/${V0}{0,1,2}
+TEST $CLI volume set $V0 cluster.self-heal-daemon off
+TEST $CLI volume set $V0 cluster.entry-self-heal off
+TEST $CLI volume set $V0 cluster.data-self-heal off
+TEST $CLI volume set $V0 cluster.metadata-self-heal on
+
+TEST $CLI volume start $V0
+TEST $GFS --volfile-id=$V0 --volfile-server=$H0 $M0;
+TEST $GFS --volfile-id=$V0 --volfile-server=$H0 $M1;
+dd if=/dev/zero of=$M0/datafile bs=1024 count=1024
+EXPECT "^0$" echo $?
+
+#Perform a lookup from second client to validate the metadata
+
+TEST ls $M1/datafile
+
+#Check the last 10 lines of log entries
+metadataHealCount=`tail -10 $LOGDIR/mnt-glusterfs-1.log| grep "performing metadata selfheal" | wc -l`
```

Comment:
@itisravi I think the log files are not deleted for every ".t" test file. If that is true, then an earlier test case may have a client-side metadata heal that could be part of the grep. So I added the tail to avoid this.

@itisravi Is there any way to identify a client-side heal other than the logs?

Reply:
> @itisravi I think the log files are not deleted for every ".t" test file. If that is true, then an earlier test case may have a client-side metadata heal that could be part of the grep. So I added the tail to avoid this.

https://build.gluster.org/job/gh_centos7-regression/2185/consoleFull says `Logs preserved in tarball....` for every .t . So I guess they are archived + cleaned up before running every new .t Otherwise no point in archiving individual .t results. @rakshithakamath94 , would you know if we delete the previous logs before running the next .t?

> @itisravi Is there any way to identify a client-side heal other than the logs?

Maybe enable profile info and check if there are inodelks, setxattrs etc.

Reply:
@itisravi I have changed the test case to check for setattr